### PR TITLE
[#108] - Add field to choose units system for connected user

### PR DESCRIPTION
**Pull request by richardcrete**

### CODE REVIEW

File: src/Entity/User.php [hunk 1]
```diff
@@ -121,6 +121,9 @@ public function __construct()
     #[ORM\Column(type: Types::STRING, length: 39, nullable: true)]
     private string $ipAddress;
 
+    #[ORM\Column(type: Types::BOOLEAN, options: ['default' => 0])]
```
Comment (florian987) [inline src/Entity/User.php:124]: Add nullable: false, it should not be null

File: src/Entity/User.php [hunk 2]
```diff
@@ -499,4 +502,16 @@ public function setIpAddress(string $ipAddress): static
 
         return $this;
     }
+
+    public function isImperial(): ?bool
```
Comment (florian987) [inline src/Entity/User.php:506]: Remove ?, it should not allow null values

File: templates/Coaster/show.html.twig [hunk 3]
```diff
@@ -10,7 +10,7 @@
     {# <meta name="description" content="{{ coaster.name }} is a {{ coaster.manufacturer.name }} roller coaster"/> #}
 {# {% endblock %} #}
 
-{% set isImperial = units.isImperial(app.request.locale) %}
+{% set isImperial = units.isImperial(app.request.locale, app.user) %}
```
Comment (florian987) [inline templates/Coaster/show.html.twig:13]: Can you remove this and inject dependencies at the service level
Example: https://stackoverflow.com/questions/39871558/how-to-get-the-current-user-in-twig-extension-service-symfony2-8/39876994#39876994

File: templates/Park/show.html.twig [hunk 4]
```diff
@@ -5,7 +5,7 @@
 {% set subTitle = park.country.name|trans([], 'database') %}
 {% block title %}{{ title }} - {{ subTitle }}{% endblock %}
 
-{% set isImperial = units.isImperial(app.request.locale) %}
+{% set isImperial = units.isImperial(app.request.locale, app.user) %}
```
Comment (florian987) [inline templates/Park/show.html.twig:8]: To be removed when dependency injection is fixed

File: templates/Search/results.html.twig [hunk 5]
```diff
@@ -1,4 +1,4 @@
-{% set isImperial = units.isImperial(app.request.locale) %}
+{% set isImperial = units.isImperial(app.request.locale, app.user) %}
```
Comment (florian987) [inline templates/Search/results.html.twig:1]: To be removed when dependency injection is fixed

**Review by florian987 — CHANGES_REQUESTED** [review]
Thanks! Few changes suggested to optimize this and we will be able to merge 😎

File: src/Entity/User.php (hunk 1 above)
Comment (richardcrete) [reply src/Entity/User.php]: Done

File: src/Entity/User.php (hunk 2 above)
Comment (richardcrete) [reply src/Entity/User.php]: Done

File: templates/Coaster/show.html.twig (hunk 3 above)
Comment (richardcrete) [reply templates/Coaster/show.html.twig]: Made the changes. Can you check?

File: templates/Park/show.html.twig (hunk 4 above)
Comment (richardcrete) [reply templates/Park/show.html.twig]: Same

File: templates/Search/results.html.twig (hunk 5 above)
Comment (richardcrete) [reply templates/Search/results.html.twig]: Same

**Review by florian987 — APPROVED** [review]
parfait! thanks

**Pull request merged**